### ISSUE TITLE
Enhancement: Favorite symbol palette on ctrl & left-click

### ISSUE THREAD

**Issue by mlerjen** [OpenOrienteering/mapper#1185]
### Steps to reproduce/ Actual behaviour
1. Drawing the map from the  scribble notes on the desktop, one hand is on the mouse to draw and switch tools, while the other is on the keyboard to press special keys (Crtl, Space a.s.o.) and short-cuts (as crtl&G, K ,O or E a.s.o.) to switch tool and execute most of the operations without using toolbars neither menues. This makes it possible to stay "on-site" with the mouse cursor most of the time.
2. The only bump in the drawing workflow occurs all these times you have to leave the main window to change the current symbol for drawing/replacing/ filling. 
[I usually draw big contiguous vegetation patch-works with a helper area symbol, then cut it into the patches and then change them to the according intended symbol]

### Expected behaviour
1. It would much smoother to apply the same concept with the ring palette appearing at left-click also for  a set of nine favorite symbols. Thus on crtl & left-click an alternate ring palette appears with nine symbols to switch to. These symbol could be somehow defined and marked in the Symbol dialog (add to favorites).
2. That way, the mouse cursor much more seldom had to leave the main window, but would remain on the site of drawing action.

### Configuration
Mapper Version: 0.8.3
Operating System: Windows 7

**Comment by mlerjen** [comment]
![image](https://user-images.githubusercontent.com/15611653/49218526-eb3a1700-f3d0-11e8-98eb-02caa0747e3c.png)
fig.1. use case...

**Comment by mlerjen** [comment]
This image show the mouse paths it took me to draw the building in the background.

The movement towards tool buttons and drop-down menues is very reduced due my use of keyboard shortcuts.

The remaining bump really still is the selection of the symbols. What I tried here was to at least pull my favourites to the left end of the symbol window. That is better than nothing but... if I had my favourites on the context menu... :1st_place_medal: 

![IOGraphica - 21 minutes (from 16-07 to 16-28)](https://user-images.githubusercontent.com/15611653/149421225-8c662955-1fa1-447b-8333-bb7e4e18ecc1.png)

**Comment by ollesmaps** [comment]
Relates to #948.

**Comment by dl3sdo** [comment]
I assume you want something like this:

https://github.com/OpenOrienteering/mapper/assets/14225666/c76c28c5-c0b0-487e-a858-dbc411d0b51c

**Comment by mlerjen** [comment]
yes. X-)

**Comment by mlerjen** [comment]
Imagine alt & right-click would open a wheel  menu in which  you could switch between template configurations. :-)

**Comment by mlerjen** [comment]
Yes. Similar to this one https://github.com/OpenOrienteering/mapper/issues/534 -> comment#8

**Comment by mlerjen** [comment]
Or just put blanks  to fill up to six?

Or just fill it with the six "recent" symbols? (That way you could work in full "recent" mode by defining zero favorites).

Are 6 favorites enough?  Would it be possible to stack a second ctrl-right click on top of the first palette to get a second palette with 6 more "favorites" resp. "recents"?

**Comment by dl3sdo** [comment]
I now use blanks to fill up to six symbols.
Up to 8 symbols can be shown without overlap of the symbols (and without increasing the size of the symbol wheel menu).
@mlerjen: which number of symbols would you consider as sufficient for the symbol wheel menu? 

I thought as well about a mechanism to use the most used symbols as default.
The user could still be able to select symbols to make them sticky, but that might become superfluous indeed.

https://github.com/OpenOrienteering/mapper/assets/14225666/1444c521-7f0f-46f8-b44f-c6b3bee88728

**Comment by mlerjen** [comment]
Very nice.
I would propose to test the feature, as it is all about developing the smooth work flow.

Some thoughts beforehand: 
- There is always the option to pick a symbol directly from the map window by selecting a drawn object with this symbol you want. Thus to fill the palette with the most popular symbols might turn it useless. You could pick the symbol from the map window most of the time. Actually this is also a point against working with "most recent".
- The palette is therefore more to pick up the "rare but regular" ??
- Now, in some cases you don't know which symbols are available on the palette. Would you then check the palette and if no return and follow the traditional path by the symbol window? You must be somehow assured that the first path is the path to achieve your goal.... How about a second ctrl&right click on the palette opened up the floating symbol window at the point you click? You'd have just two clicks -> palette -> symbolwindow to get to the symbol you want. There maybe use a sort by used/not used and then sort by number such as the used symbols shine on top but in the order we got used to.

- There is another use case though, where the palette is very handy indeed. It is when you switch symbol or fill with another symbol. In that case you can not select the final symbol from the map window but will have to pick it up from the symbol window instead. e.g. in the drawing areas workflow I use (draw common outline, cut along borders, assign symbol)...
![image](https://github.com/OpenOrienteering/mapper/assets/15611653/4721283f-27be-48dd-9787-8b60028e7af4)

**Comment by ollesmaps** [comment]
I would be very happy if this works also in Android. The symbol icons could appear in the same place as colors of Paint-on-template . See #948
